internal/storage/bucket_handle.go

```diff
@@ -96,3 +98,38 @@ func (b *bucketHandle) StatObject(ctx context.Context, req *gcs.StatObjectReques
 
 	return
 }
+
+func (b *bucketHandle) CopyObject(ctx context.Context, req *gcs.CopyObjectRequest) (o *gcs.Object, err error) {
+	srcObj := b.bucket.Object(req.SrcName)
+	dstObj := b.bucket.Object(req.DstName)
+
+	// Switching to the requested generation of source object.
+	if req.SrcGeneration != 0 {
+		srcObj = srcObj.Generation(req.SrcGeneration)
+	}
+
+	// Putting a condition that the metaGeneration of source should match *req.SrcMetaGenerationPrecondition for copy operation to occur.
+	if req.SrcMetaGenerationPrecondition != nil {
+		srcObj = srcObj.If(storage.Conditions{MetagenerationMatch: *req.SrcMetaGenerationPrecondition})
+	}
+
+	objAttrs, err := dstObj.CopierFrom(srcObj).Run(ctx)
+
+	if err != nil {
+		switch ee := err.(type) {
+		case *googleapi.Error:
+			if ee.Code == http.StatusPreconditionFailed {
+				err = &gcs.PreconditionError{Err: ee}
+			}
+			if ee.Code == http.StatusNotFound {
+				err = &gcs.NotFoundError{Err: storage.ErrObjectNotExist}
+			}
+		default:
+			err = fmt.Errorf("Error in copying object: %v", err)
```

Comment:
Error info should always start with lowercase letter. Use %w instead of %v.
Ref: https://stackoverflow.com/questions/61283248/format-errors-in-go-s-v-or-w#:~:text=Use%20%25w%20instead%20of%20%25v,later%20be%20unwrapped%20with%20errors.

Reply:
fixed